### ISSUE TITLE
contenteditable

### ISSUE THREAD

**Issue by jgraham** [web-platform-tests/interop#188]
### Description

Alignment on apsects of contentEditable that have been agreed as part of the Interop 2022 [investigation](https://github.com/web-platform-tests/interop-2022-editing/issues/5).

### Rationale

contenteditable interop has been a longstanding issue on the web, and causes significant difficulties for users and authors as libraries need to handle browser differences, often using complex approaches like DOM diffing. However in some cases even the few invariants assumed by libraries are broken, and so there can be user-facing breakage.

The lack of a trustworthy specification, and constraints imposed by deployed content makes it hard to fully align on contenteditable behaviour. However the interop 2022 investigation revealed some areas in which browsers could better align, and in doing so make it easier to write working editing libraries.

### Tests

https://github.com/web-platform-tests/wpt/blob/master/editing/other/insertparagraph-in-inline-editing-host.tentative.html

**Comment by foolip** [comment]
Is https://w3c.github.io/contentEditable/ part of the spec for this?

**Comment by foolip** [comment]
`contenteditable` was mentioned by WordPress developer @youknowriad in https://make.wordpress.org/core/2022/10/10/seeking-proposals-for-interop-2023/#comment-43928:

> There’s a lot of differences and bugs with how Selection and contenteditables work. A lot of these are already “hacked” in RichText. @ellatrix would know more here.

Thanks Riad for sharing!

**Comment by jgraham** [comment]
The spec situation here is still A Mess. We're hoping that following the 2022 Investigate item we have consensus on the behaviour in the mentioned tests, and that is sufficient to score some progress in the most impactful areas of implementation consistency in 2023.

**Comment by tbondwilkinson** [comment]
contenteditable is a P4 for Google's [closure library](https://github.com/google/closure-library)

"This could imapct goog.editor, which (as far as I can tell) is critical infrastructure that nobody really understands."

**Comment by gsnedders** [comment]
> The spec situation here is still A Mess. We're hoping that following the 2022 Investigate item we have consensus on the behaviour in the mentioned tests, and that is sufficient to score some progress in the most impactful areas of implementation consistency in 2023.

Is the hope the group will waive the requirement for a standards track document here?

**Comment by jgraham** [comment]
> Is the hope the group will waive the requirement for a standards track document here?

I hope so.

I don't know that it's explicitly stated anywhere but it seems unreasonable to take the position that browsers are OK to ship a feature but aren't OK to accept proposals to improve the interoperability of that feature.

Of course there are often additional considerations to fixing up these legacy features, but that's why we had an Investigate effort in 2022 before making a focus area proposal.

**Comment by foolip** [comment]
In the [MDN short survey on APIs & JavaScript](https://github.com/web-platform-tests/interop/issues/246), "Editing (`contenteditable`)" was selected by ~18% of survey takers, putting it in the middle third of options. (There is [some uncertainty](https://github.com/web-platform-tests/interop/issues/246#issuecomment-1312098410) as with any survey data.)

**Comment by jensimmons** [comment]
Could you summarize the concrete proposal here? It's to add a set of tests to Interop 2023? Which tests? Do those tests represent the current standard? Or perhaps instead an Investigation project sense of what a standards should be in the future?

**Comment by jgraham** [comment]
I've updated the initial comment, it's one test.

The contenteditable standard remains a mess, but I believe there was agreement in the investigation project that test represents behaviour we should converge on.

**Comment by jensimmons** [comment]
Thanks @jgraham. So it's this list: https://wpt.fyi/results/editing/other?label=master&label=experimental&aligned&view=subtest&q=insertparagraph-in-inline-editing-host.tentative.html

**Comment by nairnandu** [comment]
Thank you for proposing contenteditable for inclusion in Interop 2023.

We are pleased to let you know that this proposal was accepted as part of the Web Compat 2023 focus area. You can follow the progress of this Focus Area on the [Interop 2023 dashboard](https://wpt.fyi/interop-2023).

For an overview of our process, see the [proposal selection summary](https://github.com/web-platform-tests/interop/blob/main/2023/proposal-selection.md). Thank you for contributing to Interop 2023!

_Posted on behalf of the Interop team._
